container-disc/src/main/java/com/yahoo/container/jdisc/component/Deconstructor.java

```diff
@@ -52,27 +55,44 @@ public void deconstruct(Object component) {
     }
 
     private static class DestructComponentTask implements Runnable {
+
         private final AbstractComponent component;
 
         DestructComponentTask(AbstractComponent component) {
             this.component = component;
         }
 
+        /** Returns a random value which will be different across identical containers invoking this at the same time */
+        private long random() {
+            return new SecureRandom().nextLong();
+        }
+
         public void run() {
             log.info("Starting deconstruction of " + component);
             try {
                 component.deconstruct();
                 log.info("Finished deconstructing " + component);
-            } catch (Error e) {
-                try {
-                    Thread.sleep((long) (new Random(System.nanoTime()).nextDouble() * 180 * 1000));
-                } catch (InterruptedException exception) { }
-                com.yahoo.protect.Process.logAndDie("Error when deconstructing " + component, e);
-            } catch (Exception e) {
+            }
+            catch (Exception | NoClassDefFoundError e) { // May get class not found due to it being already unloaded
                 log.log(WARNING, "Exception thrown when deconstructing " + component, e);
-            } catch (Throwable t) {
-                log.log(WARNING, "Unexpected Throwable thrown when deconstructing " + component, t);
+            }
+            catch (Error e) {
+                try {
+                    // Randomize restart over 10 minutes to avoid simultaneous cluster restarts
+                    long randomSleepSeconds = random() * 60 * 10;
+                    log.log(LogLevel.FATAL, "Error when deconstructing " + component + " Will sleep for " +
```

Comment:
Nitpicking, but you forgot to put a '.' after the component.  `(". Will sleep...)`

Reply:
Fixed.